NFIQ2/NFIQ2Algorithm/src/nfiq2/modelinfo.cpp

```diff
@@ -0,0 +1,124 @@
+#include <nfiq2/modelinfo.hpp>
+#include <nfiq2/nfiqexception.hpp>
+
+#include <fstream>
+#include <iostream>
+#include <string>
+
+const std::string NFIQ::ModelInfo::ModelInfoKeyName = "Name";
+const std::string NFIQ::ModelInfo::ModelInfoKeyTrainer = "Trainer";
+const std::string NFIQ::ModelInfo::ModelInfoKeyDescription = "Description";
+const std::string NFIQ::ModelInfo::ModelInfoKeyVersion = "Version";
+const std::string NFIQ::ModelInfo::ModelInfoKeyPath = "Path";
+const std::string NFIQ::ModelInfo::ModelInfoKeyHash = "Hash";
+
+NFIQ::ModelInfo::ModelInfo(const std::string &modelInfoFilePath)
+{
+	std::unordered_map<std::string, std::string> modelInfoMap {};
+	try {
+		modelInfoMap = parseModelInfoFile(modelInfoFilePath);
+	} catch (...) {
+		throw NFIQ::NFIQException(1, "Model Class Failed Construction");
+	}
+
+	this->modelName = modelInfoMap[ModelInfoKeyName];
+	this->modelTrainer = modelInfoMap[ModelInfoKeyTrainer];
+	this->modelDescription = modelInfoMap[ModelInfoKeyDescription];
+	this->modelVersion = modelInfoMap[ModelInfoKeyVersion];
+	this->modelPath = modelInfoMap[ModelInfoKeyPath];
+	this->modelHash = modelInfoMap[ModelInfoKeyHash];
+}
+
+NFIQ::ModelInfo::~ModelInfo()
+{
+}
+
+std::string
+NFIQ::ModelInfo::getModelName() const
+{
+	return this->modelName;
+}
+
+std::string
+NFIQ::ModelInfo::getModelTrainer() const
+{
+	return this->modelTrainer;
+}
+
+std::string
+NFIQ::ModelInfo::getModelDescription() const
+{
+	return this->modelDescription;
+}
+
+std::string
+NFIQ::ModelInfo::getModelVersion() const
+{
+	return this->modelVersion;
+}
+
+std::string
+NFIQ::ModelInfo::getModelPath() const
+{
+	return this->modelPath;
+}
+
+std::string
+NFIQ::ModelInfo::getModelHash() const
+{
+	return this->modelHash;
+}
+
+void
+NFIQ::ModelInfo::setModelPath(const std::string &newPath)
+{
+	this->modelPath = newPath;
+}
+
+std::unordered_map<std::string, std::string>
+NFIQ::parseModelInfoFile(const std::string &modelInfoFilePath)
+{
+	std::unordered_map<std::string, std::string> modelInfoMap {};
+
+	std::ifstream fp(modelInfoFilePath);
+	std::string line;
+
+	if (fp.is_open()) {
+
+		while (std::getline(fp, line)) {
+			if (line.find('=') == std::string::npos) {
+				continue;
+			} else {
+				const std::string start = line.substr(
+				    0, line.find('=') - 1);
+				const std::string end = line.substr(
+				    line.find('=') + 2, line.size() - 1);
```

Comment:
Sanity check that there's room in the string to do this substr (e.g., malformed lines `Key =` or `= Value`)